### PR TITLE
fix: Gallery View delete using expanded form

### PR DESCRIPTION
**Pull request by rohittp0**
## Change Summary

Fixed record not found error when deleting using expanded form of Gallery View
## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [x] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

### CODE REVIEW

**Review by dstala — APPROVED** [review]

**Comment by dstala** [comment]
Node: **v18.19.1**
Arch: **arm64**
Platform: **darwin**
Docker: **false**
RootDB: **pg**
PackageVersion: **0.207.2**

In grid + groupBy, triggering bulk delete for a record keeps loading forever. record is deleted on page reload
@rohittp0

File: packages/nc-gui/components/smartsheet/grid/GroupByTable.vue [hunk 1]
```diff
@@ -136,6 +136,14 @@ const pagination = computed(() => {
     extraStyle: `margin-left: ${scrollBump.value}px;`,
   }
 })
+
+async function deleteSelectedRowsWrapper() {
+  if (!deleteSelectedRows) throw new Error('Feature not implemented')
+
+  await deleteSelectedRows()
+  // reload table data
+  await reloadTableData({ shouldShowLoading: true })
```
Comment (mertmit) [inline packages/nc-gui/components/smartsheet/grid/GroupByTable.vue:145]: What is the expected UX here @dstala, throwing will cause error modal to be shown and make UI unusable until refresh if I am not missing something

File: packages/nc-gui/composables/useData.ts [hunk 2]
```diff
@@ -732,7 +732,7 @@ export function useData(args: {
 
     await callbacks?.syncCount?.()
     await callbacks?.syncPagination?.()
-    await callbacks?.globalCallback?.()
+    callbacks?.globalCallback?.()
```
Comment (mertmit) [inline packages/nc-gui/composables/useData.ts:735]: What is the purpose of this change? This is a shared composable so is changing callback logic a good idea?

File: packages/nc-gui/components/smartsheet/grid/GroupByTable.vue (hunk 1 above)
Comment (rohittp0) [reply packages/nc-gui/components/smartsheet/grid/GroupByTable.vue:145]: It should not throw. the `deleteSelectedRows` should always be set. The check is there just to get rid of TS warning.

File: packages/nc-gui/composables/useData.ts (hunk 2 above)
Comment (rohittp0) [reply packages/nc-gui/composables/useData.ts:735]: The type definition of `globalCallback` is `globalCallback?: (...args: any[]) => void`. It is not async. Removed the `await` to handle TS warning. I think the `await` was there by mistake.

Comment (mertmit) [reply packages/nc-gui/composables/useData.ts:735]: Please keep await and replace type definition to expect a promise instead, having await on a non-promise will not have any significant difference but counter-part can cause unexpected results (as we await wrapper function everything should be handled before going to next step)

File: packages/nc-gui/components/smartsheet/grid/GroupByTable.vue (hunk 1 above)
Comment (mertmit) [reply packages/nc-gui/components/smartsheet/grid/GroupByTable.vue:145]: Please use following as this is the common pattern we follow
```
if (!deleteSelectedRows) {
  return
}
```

**Review by mertmit — APPROVED** [review]
